js/modules/k6/experimental/fs/module.go

```diff
@@ -0,0 +1,185 @@
+// Package fs provides a k6 module that allows users to interact with files from the
+// local filesystem.
+package fs
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/dop251/goja"
+	"go.k6.io/k6/js/common"
+	"go.k6.io/k6/js/modules"
+	"go.k6.io/k6/js/promises"
+	"go.k6.io/k6/lib/fsext"
+)
+
+type (
+	// RootModule is the global module instance that will create instances of our
+	// module for each VU.
+	RootModule struct {
+		fileRegistry fileRegistry
+	}
+
+	// ModuleInstance represents an instance of the fs module for a single VU.
+	ModuleInstance struct {
+		vu modules.VU
+
+		fileRegistry *fileRegistry
+	}
+)
+
+var (
+	_ modules.Module   = &RootModule{}
+	_ modules.Instance = &ModuleInstance{}
+)
+
+// New returns a pointer to a new [RootModule] instance.
+func New() *RootModule {
+	return &RootModule{
+		fileRegistry: *newFileRegistry(),
+	}
+}
+
+// NewModuleInstance implements the modules.Module interface and returns a new
+// instance of our module for the given VU.
+func (rm *RootModule) NewModuleInstance(vu modules.VU) modules.Instance {
+	return &ModuleInstance{vu: vu, fileRegistry: &rm.fileRegistry}
+}
+
+// Exports implements the modules.Module interface and returns the exports of
+// our module.
+func (mi *ModuleInstance) Exports() modules.Exports {
+	return modules.Exports{
+		Named: map[string]any{
+			"open":     mi.Open,
+			"File":     File{},
```

Comment:
So that users can `instanceof` over the result of `open` and `stat`, see https://github.com/grafana/k6/pull/3165#discussion_r1254231656

Reply:
I'm afraid that `instanceof` won't work here since, in that case, the export `File` is an object (go's `File{}`), not the class. Maybe I'm wrong here, but I believe it is worth checking.

Reply:
I looked into it, and you're right, it doesn't work as I expected. I dug a bit further into it, and making it work the way I would like: `fileInfo instanceof FileInfo` looks quite tricky at the moment; or at least I couldn't make sense of how to make it work 😞